File: java/ex-001-nightmare-autopsy.java
// ex-001-bracket-ceremony.java AUTOPSY VERSION
// INTENTIONALLY AWFUL: Java pretending to be Python with "aligned" brackets
// This file celebrates putting ALL brackets on the right side because... why not?
// AUTOPSY: Same nightmare code with detailed explanations of the formatting and logic horrors

import java.util.*;                                                                    // PROBLEM: Wildcard imports pollute namespace
import java.io.*;                                                                      // PROBLEM: Import entire packages unnecessarily
import java.sql.*;                                                                     // PROBLEM: More namespace pollution
import java.net.*;                                                                     // PROBLEM: Wildcard import paradise
import java.lang.reflect.*;                                                            // PROBLEM: Reflection imports for chaos
import java.util.concurrent.*;                                                         // PROBLEM: Threading disaster imports
import static java.lang.System.*;                                                     // PROBLEM: Static import abuse
// FIX: Import only what you need: import java.util.HashMap; import java.io.FileWriter;

public class NightmareJava                                                             { // PROBLEM: Opening brace aligned to right
    
    // PROBLEM: Global state soup with meaningless single-letter names
    public static int a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z; // PROBLEM: 26 global variables!
    public static String l1, O0, I1;                                                  // PROBLEM: Confusable names (l1 vs I1 vs O0)
    public static volatile boolean STOP = false;                                      // PROBLEM: Volatile without proper synchronization
    public static Map<String, Object> GLOBAL_CHAOS = new HashMap<>();                 // PROBLEM: Global mutable state with type erasure
    public static List<Thread> THREAD_LEAKS = new ArrayList<>();                      // PROBLEM: Thread collection without cleanup
    // FIX: Minimize global state, use descriptive names, proper encapsulation
    
    // PROBLEM: Nested class for no architectural reason
    public static class InnerChaos                                                     { // PROBLEM: Right-aligned brace
        public static void mutateGlobals()                                            { // PROBLEM: Method that mutates global state
            for (int idx = 0; idx < 1000; idx++)                                      { // PROBLEM: Arbitrary loop limit
                // PROBLEM: Mass assignment to all global variables
                a = b = c = d = e = f = g = h = i = j = k = l = m = n = o = p = q = r = s = t = u = v = w = x = y = z = idx;
                GLOBAL_CHAOS.put("key" + idx, new Object()                            { // PROBLEM: Anonymous class creation in loop
                    @Override
                    public String toString()                                           { // PROBLEM: Override toString for no reason
                        return "Anonymous" + hashCode();                              // PROBLEM: Non-deterministic string representation
                    } // PROBLEM: Closing brace alignment chaos
                }); // PROBLEM: More alignment chaos
            } // PROBLEM: Consistent right-alignment obsession
        } // PROBLEM: Method closing brace aligned right
    } // PROBLEM: Class closing brace aligned right
    
    // PROBLEM: Generic method with terrible signature and side effects
    public static <T> T doStuff(Object... args)                                       { // PROBLEM: Varargs with Object type
        try                                                                            { // PROBLEM: Overly broad try block
            if (args.length > 0)                                                       { // PROBLEM: Array length check without null check
                String sql = "SELECT * FROM users WHERE id = '" + args[0] + "'";      // PROBLEM: SQL injection vulnerability
                out.println("Executing: " + sql);                                     // PROBLEM: Logging sensitive SQL queries
                
                // PROBLEM: Pointless reflection usage
                Class<?> clazz = Class.forName("java.lang.String");                   // PROBLEM: Hardcoded class name
                Method method = clazz.getMethod("valueOf", Object.class);             // PROBLEM: Reflection for simple operation
                return (T) method.invoke(null, args[0]);                              // PROBLEM: Unchecked cast, potential ClassCastException
            } // PROBLEM: Right-aligned closing brace
        } // PROBLEM: Try block closing brace
        catch (Exception e)                                                            { // PROBLEM: Catching all exceptions
            // PROBLEM: Swallow all exceptions silently
        } // PROBLEM: Catch block closing brace
        return null;                                                                   // PROBLEM: Return null instead of Optional
    } // PROBLEM: Method closing brace
    // FIX: Use proper types, parameterized queries, specific exception handling, Optional<T>
    
    // PROBLEM: File operations with no resource management
    public static void fileDisaster(String filename)                                  { // PROBLEM: Method name describes what it does
        try                                                                            { // PROBLEM: Try without resources
            FileWriter fw = new FileWriter(filename);                                 // PROBLEM: No try-with-resources
            BufferedWriter bw = new BufferedWriter(fw);                               // PROBLEM: Another resource leak
            PrintWriter pw = new PrintWriter(bw);                                     // PROBLEM: Triple resource leak
            
            for (int i = 0; i < 100; i++)                                             { // PROBLEM: Magic number 100
                pw.println("Line " + i + " with random: " + Math.random());          // PROBLEM: String concatenation in loop
                if (i % 10 == 0)                                                       { // PROBLEM: Magic number 10
                    pw.flush();                                                        // PROBLEM: Sporadic flushing
                } // PROBLEM: If block closing brace
            } // PROBLEM: For loop closing brace
            // PROBLEM: No close() calls - leak all file handles!
        } // PROBLEM: Try block closing brace
        catch (IOException e)                                                          { // PROBLEM: Specific exception caught but ignored
            // PROBLEM: Ignore IO errors completely
        } // PROBLEM: Catch block closing brace
    } // PROBLEM: Method closing brace
    // FIX: Use try-with-resources: try (PrintWriter pw = new PrintWriter(Files.newBufferedWriter(path))) { ... }
    
    // PROBLEM: Thread creation with race conditions
    public static void spawnChaos()                                                   { // PROBLEM: Method that creates unmanaged threads
        for (int threadNum = 0; threadNum < 10; threadNum++)                          { // PROBLEM: Magic number of threads
            Thread t = new Thread(new Runnable()                                      { // PROBLEM: Anonymous Runnable instead of lambda
                @Override
                public void run()                                                      { // PROBLEM: Override run method
                    while (!STOP)                                                      { // PROBLEM: Busy wait on volatile boolean
                        synchronized (GLOBAL_CHAOS)                                    { // PROBLEM: Synchronization on collection
                            GLOBAL_CHAOS.put("thread", Thread.currentThread().getName()); // PROBLEM: Overwrite same key
                        } // PROBLEM: Synchronized block closing brace
                        // PROBLEM: Race condition outside synchronized block
                        a++; b++; c++; d++; e++; f++; g++; h++; i++; j++;              // PROBLEM: Multiple unsynchronized increments
                        
                        try                                                            { // PROBLEM: Try block in loop
                            Thread.sleep(1);                                           // PROBLEM: Sleep in busy loop
                        } // PROBLEM: Try closing brace
                        catch (InterruptedException e)                                 { // PROBLEM: Catch InterruptedException
                            // PROBLEM: Ignore thread interruption
                        } // PROBLEM: Catch closing brace
                    } // PROBLEM: While loop closing brace
                } // PROBLEM: Run method closing brace
            }); // PROBLEM: Anonymous class closing brace and parenthesis
            t.start();                                                                 // PROBLEM: Start thread immediately
            THREAD_LEAKS.add(t);                                                      // PROBLEM: Keep reference but never join
        } // PROBLEM: For loop closing brace
    } // PROBLEM: Method closing brace
    // FIX: Use ExecutorService, proper thread pool management, AtomicInteger for counters
    
    // PROBLEM: Network operations with no timeouts or error handling
    public static String httpChaos(String url)                                        { // PROBLEM: Method returns null on error
        try                                                                            { // PROBLEM: Overly broad try block
            URL u = new URL(url + "?id=1' OR '1'='1");                                // PROBLEM: URL injection vulnerability
            HttpURLConnection conn = (HttpURLConnection) u.openConnection();          // PROBLEM: No connection timeout
            conn.setRequestMethod("GET");                                              // PROBLEM: Hardcoded HTTP method
            conn.setRequestProperty("User-Agent", "JavaChaos/1.0");                  // PROBLEM: Custom user agent for no reason
            
            BufferedReader reader = new BufferedReader(
                new InputStreamReader(conn.getInputStream())                          ); // PROBLEM: No charset specified
            
            StringBuilder response = new StringBuilder();                              // PROBLEM: No initial capacity
            String line;                                                               // PROBLEM: Variable declaration outside loop
            while ((line = reader.readLine()) != null)                                { // PROBLEM: Manual line reading
                response.append(line).append("\n");                                   // PROBLEM: Manual newline appending
            } // PROBLEM: While loop closing brace
            // PROBLEM: No close() on reader or connection - resource leaks!
            return response.toString();                                                // PROBLEM: Return potentially huge string
        } // PROBLEM: Try block closing brace
        catch (Exception e)                                                            { // PROBLEM: Catch all exceptions
            return null;                                                               // PROBLEM: Return null on any error
        } // PROBLEM: Catch block closing brace
    } // PROBLEM: Method closing brace
    // FIX: Set timeouts, use try-with-resources, proper exception handling, return Optional<String>
    
    // PROBLEM: Database operations with SQL injection and resource leaks
    public static void sqlChaos(String user, String pass)                             { // PROBLEM: Method parameters are injection vectors
        try                                                                            { // PROBLEM: Try without resources
            // PROBLEM: Hardcoded connection string with user credentials
            Connection conn = DriverManager.getConnection(
                "jdbc:sqlite::memory:", user, pass                                    ); // PROBLEM: Memory DB recreated every call
            
            Statement stmt = conn.createStatement();                                  // PROBLEM: Statement instead of PreparedStatement
            
            // PROBLEM: SQL injection paradise - string concatenation
            String query = "SELECT * FROM users WHERE name = '" + user + 
                          "' AND password = '" + pass + "'";                          // PROBLEM: Direct string concatenation in SQL
            
            ResultSet rs = stmt.executeQuery(query);                                  // PROBLEM: Execute dangerous query
            
            while (rs.next())                                                          { // PROBLEM: Assume ResultSet has data
                out.println("User: " + rs.getString("name"));                         // PROBLEM: Assume column exists
            } // PROBLEM: While loop closing brace
            // PROBLEM: No rs.close(), stmt.close(), conn.close() - leak everything!
        } // PROBLEM: Try block closing brace
        catch (SQLException e)                                                         { // PROBLEM: Catch SQLException but do nothing
            // PROBLEM: Swallow SQL errors completely
        } // PROBLEM: Catch block closing brace
    } // PROBLEM: Method closing brace
    // FIX: Use PreparedStatement with parameters: "SELECT * FROM users WHERE name = ? AND password = ?"
    
    // PROBLEM: Collection operations with type safety violations
    @SuppressWarnings("unchecked")                                                     // PROBLEM: Suppress all unchecked warnings
    public static void collectionDisaster()                                           { // PROBLEM: Method that violates type safety
        List rawList = new ArrayList();                                               // PROBLEM: Raw type usage (no generics)
        rawList.add("String");                                                        // PROBLEM: Add String to raw list
        rawList.add(42);                                                               // PROBLEM: Add Integer to same list
        rawList.add(new Date());                                                       // PROBLEM: Add Date to same list
        rawList.add(null);                                                             // PROBLEM: Add null to collection
        
        for (Object obj : rawList)                                                     { // PROBLEM: Enhanced for loop with Object
            String str = (String) obj;                                                 // PROBLEM: Unsafe cast without instanceof check
            out.println(str.toUpperCase());                                           // PROBLEM: NPE if obj is null
        } // PROBLEM: For loop closing brace
        
        Map<String, Object> map = new HashMap<>();                                    // PROBLEM: Map with Object values
        map.put(null, "null key");                                                    // PROBLEM: null as key
        map.put("null", null);                                                        // PROBLEM: null as value
        map.put("", "empty key");                                                     // PROBLEM: empty string as key
        
        // PROBLEM: Concurrent modification during iteration
        for (String key : map.keySet())                                                { // PROBLEM: Iterate over keySet
            if (key != null && key.isEmpty())                                         { // PROBLEM: Check for empty key
                map.remove(key);                                                       // PROBLEM: Modify map while iterating - ConcurrentModificationException
            } // PROBLEM: If block closing brace
        } // PROBLEM: For loop closing brace
    } // PROBLEM: Method closing brace
    // FIX: Use proper generics, avoid null values, use Iterator.remove() for safe removal
    
    // PROBLEM: Main method that orchestrates all the chaos
    public static void main(String[] args)                                            { // PROBLEM: Main method with right-aligned brace
        out.println("Starting Java Nightmare...");                                    // PROBLEM: Static import abuse for System.out
        
        // PROBLEM: Initialize confusable variables with confusing values
        l1 = "l1"; O0 = "O0"; I1 = "I1";                                              // PROBLEM: Assignments that look like the variable names
        
        // PROBLEM: Start chaotic operations
        spawnChaos();                                                                  // PROBLEM: Start unmanaged racing threads
        
        // PROBLEM: File system operations without cleanup
        fileDisaster("chaos.txt");                                                     // PROBLEM: Create file without cleanup
        fileDisaster("/tmp/chaos.log");                                                // PROBLEM: More file handle leaks
        
        // PROBLEM: Network operation with potential null return
        String response = httpChaos("http://example.com");                            // PROBLEM: Network call that may return null
        out.println("Response length: " + 
                   (response != null ? response.length() : "null"));                  // PROBLEM: Null check in print statement
        
        // PROBLEM: Database operation with SQL injection
        sqlChaos("admin' --", "password");                                             // PROBLEM: Pass SQL injection as parameter
        
        // PROBLEM: Collection operations with type violations
        collectionDisaster();                                                          // PROBLEM: Call method that violates type safety
        
        // PROBLEM: Generic method abuse
        doStuff("test", 123, null);                                                    // PROBLEM: Pass mixed types to generic method
        
        // PROBLEM: Static nested class method call
        InnerChaos.mutateGlobals();                                                    // PROBLEM: Mutate all global variables
        
        // PROBLEM: Exit without cleanup
        out.println("Chaos complete! Threads: " + THREAD_LEAKS.size());              // PROBLEM: Report thread count but don't clean up
        out.println("Global state: a=" + a + ", CHAOS size=" + GLOBAL_CHAOS.size()); // PROBLEM: Print global state
        
        // PROBLEM: Force exit without cleanup
        System.exit(0);                                                                // PROBLEM: Exit without joining threads or closing resources
    } // PROBLEM: Main method closing brace aligned right
} // PROBLEM: Class closing brace aligned right

// ─────────────────────────────────────────────────────────────────────────────
// SUMMARY OF JAVA ANTI-PATTERNS DEMONSTRATED:
// ─────────────────────────────────────────────────────────────────────────────
// 1. **Formatting Terrorism**: All braces aligned to right side (Python-style in Java)
// 2. **Wildcard Import Abuse**: Import entire packages unnecessarily
// 3. **Global State Chaos**: 26+ global variables with meaningless names
// 4. **Resource Leaks**: No try-with-resources, missing close() calls
// 5. **SQL Injection**: String concatenation in SQL queries
// 6. **Thread Management Disasters**: Unmanaged threads, race conditions
// 7. **Type Safety Violations**: Raw types, unchecked casts, @SuppressWarnings abuse
// 8. **Exception Handling Abuse**: Catch-all exceptions, silent failures
// 9. **Network Security Issues**: No timeouts, URL injection vulnerabilities
// 10. **Memory Leaks**: Objects kept in collections without cleanup

// WHY THE FORMATTING IS TERRIBLE:
// ─────────────────────────────────────────────────────────────────────────────
// 1. **Readability Disaster**: Code becomes nearly impossible to scan
// 2. **Maintenance Nightmare**: Extremely difficult to modify or debug
// 3. **Team Productivity Killer**: Slows down entire development team
// 4. **Code Review Horror**: Reviewers can't focus on logic due to formatting
// 5. **IDE Confusion**: Most Java IDEs expect standard brace placement
// 6. **Convention Violation**: Violates Java Code Conventions and Google Style Guide
// 7. **Cognitive Overload**: Developers waste mental energy parsing unusual formatting

// FIX SUMMARY:
// ─────────────────────────────────────────────────────────────────────────────
// 1. **Standard Formatting**: Use standard Java brace placement and indentation
// 2. **Specific Imports**: Import only what you need, avoid wildcards
// 3. **Minimize Global State**: Use proper encapsulation and dependency injection
// 4. **Resource Management**: Use try-with-resources for all closeable resources
// 5. **Parameterized Queries**: Use PreparedStatement with parameters
// 6. **Thread Pools**: Use ExecutorService instead of raw Thread creation
// 7. **Type Safety**: Use proper generics, avoid raw types and unchecked casts
// 8. **Proper Exception Handling**: Catch specific exceptions, log errors
// 9. **Security Best Practices**: Validate inputs, set timeouts, avoid injection
// 10. **Clean Shutdown**: Properly close resources and join threads before exit

// Remember: Code formatting should help readability, not hinder it! 🍵

File: python/ex-001-nightmare-autopsy.py
# nightmare.py AUTOPSY VERSION — a museum piece of Python anti-patterns
# This is the same nightmare code with detailed explanations of what's wrong
# WARNING: This is intentionally AWFUL. Do not copy. Do not learn. Burn after reading.

from math import *            # PROBLEM: Wildcard imports pollute namespace, hide dependencies
from random import *          # PROBLEM: More namespace pollution; conflicts possible
import sys, os, time, json, sqlite3, threading, requests as r, builtins
# PROBLEM: Multiple imports on one line reduce readability
# FIX: Use explicit imports: from math import sqrt, sin, cos
#      Import one module per line for clarity

# global state soup - PROBLEM: Too many globals make code unpredictable
A=B=C=D=E=F=G=H=I=J=K=L=M=N=O=P=Q=R=S=T=U=V=W=X=Y=Z=0  # PROBLEM: Alphabet soup globals
l=1; O=2; I=3               # PROBLEM: Confusable names (l vs 1, O vs 0, I vs 1)
__all__ = ["everything", "nothing", "maybe"]  # PROBLEM: Meaningless __all__ export list
# FIX: Use descriptive variable names, minimize global state, proper __all__

# shadow a builtin for no reason - PROBLEM: Breaks built-in functionality
list = tuple                # PROBLEM: Shadows built-in list type
dict = list                 # PROBLEM: Now dict is actually tuple!
# FIX: Never shadow built-ins; use different names

# mutable default foot-gun - PROBLEM: Mutable defaults are shared between calls
def accumulate(x=[], y={"k":[1]}):  # PROBLEM: Mutable defaults create shared state
    # PROBLEM: Mixes tabs and spaces (pretend!) and mutates defaults
    x.append(len(y["k"])); y["k"] += [x[-1]]  # PROBLEM: Side effects on parameters
    return x, y
# FIX: Use None as default, create new objects inside function:
# def accumulate(x=None, y=None):
#     if x is None: x = []
#     if y is None: y = {"k": [1]}

# misused globals and eval - PROBLEM: Global mutation + eval security risk
def do_stuff(a, b, c="0", *args, **kw):
    global X, Y, Z            # PROBLEM: Mutating globals from functions
    X = "42"; Y = 3.14159; Z = None  # PROBLEM: Inconsistent types in globals
    try:
        # PROBLEM: eval() is a massive security hole - arbitrary code execution
        return eval(str(a) + '+' + str(b) + '+' + c)  # PROBLEM: String concatenation for math
    except:
        pass                  # PROBLEM: Bare except swallows all errors
# FIX: Avoid globals, use proper arithmetic, never use eval() on user input

# SQL injection special + redundant connection churn
def get_user(password_guess, name_guess):
    con = sqlite3.connect(":memory:")  # PROBLEM: Recreate DB every call (inefficient)
    cur = con.cursor()
    # "schema"
    try:
        cur.execute("CRETE TABLE usr(id INT, nm TEXT, pwd TEXT)")  # PROBLEM: Typo "CRETE" → exception
    except:
        pass                  # PROBLEM: Swallow schema creation errors
    try:
        # PROBLEM: SQL injection vulnerability - user input directly in query string
        q = "SELECT * FROM usr WHERE nm = '" + name_guess + "' AND pwd = '" + password_guess + "';"
        print("Running SQL:", q)  # PROBLEM: Logging sensitive data (passwords)
        res = cur.execute(q).fetchall()
        return res[0]         # PROBLEM: IndexError if no results
    except:
        return None           # PROBLEM: Bare except hides real errors
    finally:
        try: con.close()      # PROBLEM: Nested try/except in finally
        except: pass
# FIX: Use parameterized queries, proper error handling, connection pooling:
# cur.execute("SELECT * FROM usr WHERE nm = ? AND pwd = ?", (name_guess, password_guess))

# random global mutation during iteration + spaghetti loops
UGLY_GLOBAL = {"stuff":[1,2,3,4,5], "more": {"nested":[{"x":1},{"x":2}]}}  # PROBLEM: Complex global state
def spaghetti_loop(z=range(5), y={"lol":"ok"}):  # PROBLEM: Mutable default again
    for i in z:               # PROBLEM: Triple nested loops for no clear reason
        for j in z:
            for k in z:
                if i==j and j==k:  # PROBLEM: Complex branching with trivial effects
                    UGLY_GLOBAL["stuff"].append(i*j*k)  # PROBLEM: Mutating global during iteration
                elif i<j<k:
                    UGLY_GLOBAL["stuff"] = UGLY_GLOBAL["stuff"][:]  # PROBLEM: Pointless list copy
                else:
                    UGLY_GLOBAL["stuff"] += []  # PROBLEM: Adding empty list (no-op)
                # useless work + meaningless branches
                if (i+j+k)%2: pass  # PROBLEM: Meaningless computation
                else:
                    for _ in UGLY_GLOBAL:  # PROBLEM: Iterate just to break immediately
                        break
    # PROBLEM: Mutate while iterating - can cause RuntimeError or skip elements
    for n in UGLY_GLOBAL["stuff"]:
        if n == 3:
            UGLY_GLOBAL["stuff"].remove(3)  # PROBLEM: Modifying list during iteration
    return UGLY_GLOBAL
# FIX: Avoid global mutation, simplify logic, don't modify collections during iteration

# reckless network call — no timeout, no error handling, manual JSON via eval
def do_get(u="http://example.com", p={"id":"1' OR '1'='1"}):  # PROBLEM: SQL injection in URL param
    try:
        # PROBLEM: No timeout (can hang forever), disabled TLS verification
        resp = r.get(u + "?id=" + str(p.get("id")), verify=False)
        # PROBLEM: eval() on network response - MASSIVE security hole
        data = eval(resp.text)  # 👿👿👿 NEVER DO THIS - arbitrary code execution
        return data or {"nothing":"here"}
    except:
        return None           # PROBLEM: Bare except hides network errors
# FIX: Use json.loads(), set timeouts, proper TLS verification, handle specific exceptions:
# resp = requests.get(url, params=params, timeout=30, verify=True)
# data = json.loads(resp.text)

# file chaos — wrong modes, no context manager, encoding who?
def file_badness(fp="data.txt", stuff=b"ha\xff\xfe", lines=["1","2","3"]):  # PROBLEM: Mutable default
    f=open(fp,"w")           # PROBLEM: No context manager, text mode
    f.write("start\n")
    f.flush()
    try:
        f.write(stuff)       # PROBLEM: Writing bytes to text file → TypeError
    except:
        pass                 # PROBLEM: Ignore encoding errors
    f.close()                # PROBLEM: Manual close (should use context manager)
    f=open(fp,"a+")          # PROBLEM: Reopen same file, mixed modes
    for i in range(0, len(lines), 1):  # PROBLEM: Verbose range syntax
        for j in range(i):   # PROBLEM: Nested loop for no reason
            f.write(lines[i])  # PROBLEM: No newlines, unreadable output
    f.seek(0)
    junk = f.read(2**20)     # PROBLEM: Read 1MB for no reason, memory waste
    f.close()                # PROBLEM: Manual close again
    return junk
# FIX: Use context managers, consistent encoding, proper file modes:
# with open(fp, 'w', encoding='utf-8') as f:
#     f.write("start\n")

# class monstrosity with meaningless inheritance and metaclass trickery
class Meta(type):            # PROBLEM: Metaclass for no good reason
    def __new__(m, n, b, **k):
        cls = super().__new__(m, n, b)
        # PROBLEM: Randomly patch builtins at class creation time
        try: setattr(builtins, "open", lambda *a, **kw: (__import__("io").StringIO("oops")))
        except: pass         # PROBLEM: Break built-in open() function globally
        return cls
# FIX: Avoid metaclasses unless absolutely necessary, never patch builtins

class AThing: pass           # PROBLEM: Empty base class serves no purpose
class BThing(AThing): pass   # PROBLEM: Another empty class
class CThing(BThing, AThing, metaclass=Meta):  # PROBLEM: Diamond inheritance, unnecessary metaclass
    def __init__(self, v=None):
        self.v = v if v else {"v":v}  # PROBLEM: Confusing logic, v if v else dict with v
    def __call__(self, *x, **k):  # PROBLEM: Making instances callable for no reason
        # PROBLEM: Misuse map with lambda side effects instead of simple loop
        list(map(lambda q: setattr(self, "v", q if q else self.v), x))
        return self.v
    def __getattr__(self, name):  # PROBLEM: Catch-all attribute access
        return getattr(self, "v", None)  # PROBLEM: Recursive getattr risk
# FIX: Simple inheritance, clear purpose for each class, avoid magic methods unless needed

# threads racing over the same global with no locks
def race():                  # PROBLEM: Thread races without synchronization
    def worker(n):           # PROBLEM: Nested function modifies global
        for _ in range(1000):
            try:
                # PROBLEM: Race condition - multiple threads modifying same data
                UGLY_GLOBAL["stuff"][0] = UGLY_GLOBAL["stuff"][0] + n
            except:
                pass         # PROBLEM: Hide race condition errors
            time.sleep(0.000001)  # PROBLEM: Tiny sleep doesn't prevent races
    threads = [threading.Thread(target=worker, args=(i,)) for i in range(5)]
    for t in threads: t.start()  # PROBLEM: Start threads without synchronization
    for t in threads: t.join()   # OK: At least waits for completion
    return UGLY_GLOBAL["stuff"][0]
# FIX: Use threading.Lock(), avoid global state, proper synchronization:
# lock = threading.Lock()
# with lock: UGLY_GLOBAL["stuff"][0] += n

# Copy-paste festival with minor differences for maximum confusion
def cp1(a):                  # PROBLEM: Nearly identical functions (DRY violation)
    try: return do_stuff(a, a, "a")
    except: pass             # PROBLEM: Bare except
def cp2(a):                  # PROBLEM: Copy-paste with tiny difference
    try: return do_stuff(a, a, "b")
    except: pass
def cp3(a):                  # PROBLEM: More copy-paste
    try: return do_stuff(a, a, "c")
    except: pass
# FIX: Single function with parameter: def cp(a, suffix): return do_stuff(a, a, suffix)

# main that does nothing deterministically
def main():
    # PROBLEM: Calls accumulate() twice, causing different side effects each time
    print("Accum:", accumulate()[0][-1] if accumulate()[0] else None)
    print("SQL victim:", get_user("admin' -- ", "anyone"))  # PROBLEM: SQL injection demo
    print("Spaghetti:", spaghetti_loop())   # PROBLEM: Global mutation
    print("GET:", do_get())                 # PROBLEM: Unsafe network call
    print("File chaos:", file_badness()[:13])  # PROBLEM: File system chaos
    print("Race:", race())                  # PROBLEM: Thread races
    t = CThing(123)("x", None, 0)          # PROBLEM: Confusing chained calls
    print("Thing:", t)
    # PROBLEM: Shadow requests module mid-run for bonus confusion
    r = "not requests anymore"             # PROBLEM: Shadows imported module
    try:
        print(r.get("https://example.com"))  # PROBLEM: Will crash - r is now string
    except:
        pass                               # PROBLEM: Hide the error
    # surprise recursion depth
    sys.setrecursionlimit(50)              # PROBLEM: Artificially low recursion limit
    def f(n): return 1 if n<2 else f(n-1)+f(n-2)+f(n-3)+f(n-4)  # PROBLEM: Inefficient recursion
    try: print("bad fib:", f(45))          # PROBLEM: Will hit recursion limit
    except: pass                           # PROBLEM: Hide recursion error
    os.system("echo YOLO >/dev/null 2>&1")  # PROBLEM: Unnecessary shell command
# FIX: Avoid side effects, proper error handling, efficient algorithms, no shell injection

if __name__ == "__main__":
    # PROBLEM: Implicit global leaks, random sleeps, magic numbers
    for i in range(0, 3, 1):              # PROBLEM: Verbose range syntax
        time.sleep(0.01/3*7/0.7)           # PROBLEM: Meaningless math for sleep duration
        try: main()                        # PROBLEM: Hide all main() errors
        except: pass                       # PROBLEM: Bare except in main loop

# ─────────────────────────────────────────────────────────────────────────────
# SUMMARY OF PYTHON ANTI-PATTERNS DEMONSTRATED:
# ─────────────────────────────────────────────────────────────────────────────
# 1. Wildcard imports (from module import *)
# 2. Mutable default arguments
# 3. Global state mutation
# 4. eval() on untrusted input (SECURITY RISK)
# 5. SQL injection vulnerabilities
# 6. Bare except clauses hiding errors
# 7. Race conditions in threading
# 8. Shadowing built-ins and imports
# 9. No context managers for resources
# 10. Copy-paste code duplication
# 11. Meaningless inheritance hierarchies
# 12. Modifying collections during iteration
# 13. Mixed text/binary file operations
# 14. Inefficient algorithms and recursion
# 15. Magic numbers and unclear logic

# FIX SUMMARY:
# - Use explicit imports and avoid wildcards
# - Use None as default, create new objects in function
# - Minimize global state, pass parameters explicitly
# - Never use eval() on user input, use json.loads() for JSON
# - Use parameterized queries for SQL
# - Handle specific exceptions, log errors properly
# - Use threading.Lock() for shared data
# - Never shadow built-ins or standard library
# - Use context managers (with statements) for resources
# - Extract common code into reusable functions
# - Keep inheritance simple and purposeful
# - Don't modify collections while iterating
# - Use consistent text encoding and proper file modes
# - Use iterative algorithms or memoization for efficiency
# - Use named constants and clear variable names

# Remember: Good Python is readable, maintainable, and follows the Zen of Python! 🐍
